### ISSUE TITLE
How to navigate between screens of different StackNavigators under SwitchNavigator?

### ISSUE THREAD

**Issue by wzup** [react-navigation/react-navigation#4555]
The configuration is as follows:

```
SwitchNavigator
    LoadingScreen: Component
    App: StackNavigator
        ProfileScreen
        DashboardScreen
    SignInFlow: StackNavigator
        LoginScreen
        LikeScreen
    SignUpFlow: StackNavigator
        SignUpScreen
        TCScreen
```

Then there is a very common pattern you see on Login screens:
> Don't have an account? Sign Up

So
1. we are in `LoginScreen` above.
2. We click a button '[Sign Up]'
3. We have to be navigated to `SignUpFlow`'s `SignUpScreen`.

We should keep in mind that we use `SwitchNavigator` as a parent for our StackNavigators.
So the state of the app/routes on navigation from StackNavigator to StackNavigator [must be reset to defaults](https://reactnavigation.org/docs/en/switch-navigator.html).
> The purpose of SwitchNavigator is to only ever show one screen at a time. By default, it does not handle back actions and it resets routes to their default state when you switch away.

The goal is to navigate from `SignUpScreen` to `LoginScreen`.

How to implement this common pattern with React Navigator with configuration above?

Environment is React Native for IOS

SO https://stackoverflow.com/questions/50985224/how-to-navigate-between-screens-of-different-stacknavigators-under-switchnavigat

**Comment by brentvatne** [comment]
what you describe would work fine and no special knowledge is required. if you think there might be something wrong, create a new issue with a mcve (https://stackoverflow.com/help/mcve) that runs on https://snack.expo.io I'd be happy to investigate

**Comment by wzup** [comment]
I'll do it in the immediate future.

Meanwhile I'm reading docs again and I found [this](https://reactnavigation.org/docs/en/navigating.html):
> _If we call this.props.navigation.navigate with a route name that we haven't defined on a stack navigator, nothing will happen. Said another way, we can only navigate to routes that have been defined on our stack navigator — we cannot navigate to an arbitrary component._

And this is exactly the case I experience with my configuration "_nothing will happen_". I have `SignUpScreen` but it lives in different StackNavigator. So _nothing happens_. No errors, no warnings in console, no navigation either. 
`StackNavigator`s do not communicate to each other, live in parallel universes, have their own scopes and do not know about each other. Am I right? No matter that they are descendants of `SwitchNavigator` HOC, they are completely unlinked:

```javascript
// In my code in SignInFlow stack
if( hasAccount )
    this.props.navigation.navigate('SignUpScreen') // this screen is in SignUpFlow stack
```

**Comment by skptricks** [comment]
Here is the simple example to create react navigation project and this will help you to navigate between screens. Link 👍 
[Navigating Between Screens or Activities Using React Navigation Library](https://www.skptricks.com/2018/10/navigating-between-screens-or-activity-in-react-navigation.html)
